pkg/kinflate/app/application.go

```diff
@@ -30,6 +30,8 @@ import (
 type Application interface {
 	// Resources computes and returns the resources for the app.
 	Resources() (resource.ResourceCollection, error)
+	// semiResources computes and returns the resources without name hash and name reference for the app
+	semiResources() (resource.ResourceCollection, error)
```

Comment:
Not a big fun of this naming. But I can't find a better one and this is private.
So I guess we can leave it as is. And see if someone in the future can find a better one.

Reply:
private functions in an interface ? Haven't seen these before. Looks like a code smell :) to me.

Reply:
[Here](https://github.com/kubernetes/kubectl/pull/373/files#diff-b38a7fa92a6de50adb2d8e890664c5fcR181) is using this method. `subapp` is an interface instead of a concrete type.